cli/cmd/deploy/deploy.go

```diff
@@ -57,6 +58,13 @@ func DeployCmd(cfg *config.Config) *cobra.Command {
 				}
 			}
 
+			tel := telemetry.NewTelemetry(cfg.Version)
+			tel.EmitDeployStart()
+			defer func() {
+				// telemetry errors shouldn't fail deploy command
+				_ = tel.Flush(ctx)
```

Comment:
The `ctx` here will be cancelled if the user interrupted the process, so no telemetry will be sent.

Maybe we should use a `context.Background()` with a deadline of 5s here?